parley/src/util.rs

```diff
@@ -14,3 +14,53 @@ pub(crate) fn nearly_eq(x: f32, y: f32) -> bool {
 pub(crate) fn nearly_zero(x: f32) -> bool {
     nearly_eq(x, 0.)
 }
+
+/// A rectangle.
+#[derive(Clone, Copy, Default, PartialEq)]
+pub struct Rect {
```

Comment:
What do you think about making this `non_exhaustive` (and removing `pub fn new`/`Default`), to gently discourage more general use. Obviously if people can get a single `Rect`, since it's `Copy`, they can easily make new ones, but it should be sufficient roadblocks to prevent people doing that.

I'd also consider renaming to something like `OutputRect` or similar. In particular, I *never* want Rust Analyzer to suggest this type when I am looking to import `Rect`.
Basically strongly saying to applications that they should standardise on a rectangle type, and it shouldn't be this one.

That would make this PR much easier for me to approve, as it makes this API much clearer.
But I'd also suggest that we can talk about this in office hours.

Reply:
The rename is the more important of the two, I agree; `BoundingBox` is a pretty good name here.

Reply:
Renamed to `BoundingBox`